src/Contracts/BaseData.php

```diff
@@ -30,7 +30,7 @@ public static function from(mixed ...$payloads): object;
     /**
      * @param \Illuminate\Support\Enumerable|array|\Illuminate\Pagination\AbstractPaginator|\Illuminate\Contracts\Pagination\Paginator|\Illuminate\Pagination\AbstractCursorPaginator|\Illuminate\Contracts\Pagination\CursorPaginator|\Spatie\LaravelData\DataCollection $items
      *
-     * @return \Spatie\LaravelData\DataCollection|\Spatie\LaravelData\CursorPaginatedDataCollection|\Spatie\LaravelData\PaginatedDataCollection
+     * @return ($items is \Illuminate\Pagination\AbstractCursorPaginator|\Illuminate\Pagination\CursorPaginator ? \Spatie\LaravelData\CursorPaginatedDataCollection : ($items is \Illuminate\Pagination\Paginator|\Illuminate\Pagination\AbstractPaginator ? \Spatie\LaravelData\PaginatedDataCollection : \Spatie\LaravelData\DataCollection))
```

Comment:
Would it be possible to add the TKey and TValue in the return types? 

Something like `\Spatie\LaravelData\DataCollection<int,static>` ?

This prevents me from having to do it myself in my Data objects

Reply:
How would this look like? I'm all in for better types so if you can send in a Pr I'll probably merge it! 😄